hash-tables/ex1/ex1.py

```diff
@@ -9,15 +9,35 @@
 def get_indices_of_item_weights(weights, length, limit):
     ht = HashTable(16)
 
-    """
-    YOUR CODE HERE
-    """
-
+    # make sure hash table is big enouogh for all weights, if necessary
+    while length > ht.capacity * 0.7:
+        ht = hash_table_resize(ht)
+    
+    # loop through weights
+    for i in range(len(weights)):
+        # take off current weight from limit
+        balance = limit - weights[i]
+        balance_index = hash_table_retrieve(ht, balance)
+        # if the balance does not exist, add that weight as a future possibility for the next check
+        if balance_index is None:
+            hash_table_insert(ht, weights[i], i)
+        # if there is a pair, return it in the correct format 
```

Comment:
Another way to achieve this logic would be by sorting the `(balance_index, i)` tuple in desc order. More at this [link](https://docs.python.org/2/howto/sorting.html#ascending-and-descending) on how to do that with `sort()` or `sorted()`.